From 8bbcb89aba18107e178d6134a38dfa01da29c7e8 Mon Sep 17 00:00:00 2001
From: yicheng shen <yicheng.shen@amlogic.com>
Date: Thu, 14 Nov 2019 19:25:26 +0800
Subject: [PATCH] hdmirx: fix dolbyvisoin 10 can not work issue [1/1]

PD#SWPL-16611

Problem:
DolbyVision VSIF not recognized

Solution:
return DV10 for old version VSIF

Verify:
tl1

Change-Id: Ie3e59933a65fae8126d44afbc90e1bfd68879764
Signed-off-by: yicheng shen <yicheng.shen@amlogic.com>
---
 drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_pktinfo.c | 1 +
 drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_wrapper.c | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_pktinfo.c b/drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_pktinfo.c
index 6c8ea92ba3b8..889e3a2db2d0 100644
--- a/drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_pktinfo.c
+++ b/drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_pktinfo.c
@@ -1417,6 +1417,7 @@ uint8_t rx_get_vsi_info(void)
 					dv_nopacket_timeout;
 				pkt->sbpkt.payload.data[0] = 0xffff;
 			}
+			ret = E_VSI_DV10;
 		} else if ((pkt->length == E_PKT_LENGTH_5) &&
 			(pkt->sbpkt.payload.data[0] & 0xffff)) {
 			rx.vs_info_details.dolby_vision = false;
diff --git a/drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_wrapper.c b/drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_wrapper.c
index d09bbb48a3fa..850809fd9258 100644
--- a/drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_wrapper.c
+++ b/drivers/amlogic/media/vin/tvin/hdmirx/hdmi_rx_wrapper.c
@@ -3028,7 +3028,7 @@ int hdmirx_debug(const char *buf, int size)
 		rx_pr("Hdmirx version0: %s\n", RX_VER0);
 		rx_pr("Hdmirx version1: %s\n", RX_VER1);
 		rx_pr("Hdmirx version2: %s\n", RX_VER2);
-		rx_pr("Hdmirx version2: %s\n", "ver.2019/11/18");
+		rx_pr("Hdmirx version2: %s\n", "ver.2019/11/20");
 		rx_pr("------------------\n");
 	} else if (strncmp(input[0], "port0", 5) == 0) {
 		hdmirx_open_port(TVIN_PORT_HDMI0);
-- 
2.24.1

